src/main/java/com/networknt/schema/AnyOfValidator.java

```diff
@@ -83,7 +83,12 @@ public AnyOfValidator(String schemaPath, JsonNode schemaNode, JsonSchema parentS
                     }
                 }
                 allErrors.addAll(errors);
-//                allErrors.add(renderMissingMatchValidationMessage(at));
+            }
+
+            if (config.isOpenAPI3StyleDiscriminators() && discriminatorContext.isActive()) {

```

Comment:
this fixes the case where validation passes, but the discriminator property value does not resolve to an alternative.